ui/actions/lighthouse/lighthouse.ts

```diff
@@ -4,172 +4,507 @@ import { revalidatePath } from "next/cache";
+    return { errors: [{ detail: String(error) }] };
+  }
+};
+
+/**
+ * Delete a lighthouse provider configuration by provider type
+ * Looks up the provider server-side, never exposes ID to client
+ */
+export const deleteLighthouseProviderByType = async (providerType: string) => {
+  try {
+    // First, get the provider by type
+    const providerResult = await getLighthouseProviderByType(providerType);
 
-    // Trigger connection check in background
-    if (updateData?.data?.id || configId) {
-      checkLighthouseConnection(configId);
+    if (providerResult.errors || !providerResult.data) {
+      return providerResult;
     }
-    revalidatePath("/");
-    return updateData;
+
+    const providerId = providerResult.data.id;
+
+    // Now delete it
+    const headers = await getAuthHeaders({ contentType: false });
+    const url = new URL(`${apiBaseUrl}/lighthouse/providers/${providerId}`);
+
+    const response = await fetch(url.toString(), {
+      method: "DELETE",
+      headers,
+    });
+
+    if (response.status === 204 || response.status === 200) {
+      revalidatePath("/lighthouse/config");
+      return { success: true };
+    }
+
+    const data = await response.json();
+    return data;
   } catch (error) {
-    console.error("[Server] Error in updateLighthouseConfig:", error);
-    return undefined;
+    console.error("[Server] Error in deleteLighthouseProviderByType:", error);
+    return { errors: [{ detail: String(error) }] };
+  }
+};
+
+/**
+ * Get lighthouse providers configuration with default models
+ * Returns only the default model for each provider to avoid loading hundreds of models
+ */
+export const getLighthouseProvidersConfig = async () => {
+  try {
+    const [tenantConfig, providers] = await Promise.all([
+      getTenantConfig(),
+      getLighthouseProviders(),
+    ]);
+
+    if (tenantConfig.errors || providers.errors) {
+      return {
+        errors: tenantConfig.errors || providers.errors,
+      };
+    }
+
+    const tenantData = tenantConfig?.data?.attributes;
+    const defaultProvider = tenantData?.default_provider || "";
+    const defaultModels = tenantData?.default_models || {};
+
+    // Filter only active providers
+    const activeProviders =
+      providers?.data?.filter((p: any) => p.attributes?.is_active) || [];
+
+    // Build provider configuration with only default models
+    const providersConfig = await Promise.all(
+      activeProviders.map(async (provider: any) => {
+        const providerType = provider.attributes.provider_type;
+        const defaultModelId = defaultModels[providerType];
+
+        // Fetch only the default model for this provider if it exists
+        let defaultModel = null;
+        if (defaultModelId) {
+          const headers = await getAuthHeaders({ contentType: false });
+          const url = `${apiBaseUrl}/lighthouse/models?filter[provider_type]=${providerType}&filter[model_id]=${defaultModelId}`;
```

Comment:
No URL encoding for defaultModelId. If model ID contains special characters, this breaks.

```suggestion
          const url = `${apiBaseUrl}/lighthouse/models?filter[provider_type]=${providerType}&filter[model_id]=${defaultModelId}`;const url = `${apiBaseUrl}/lighthouse/models?${new URLSearchParams({
    'filter[provider_type]': providerType,
    'filter[model_id]': defaultModelId
  })}`;
```